From a2e5a35d0c4b77b5da5455c92f2881fdd5e65ead Mon Sep 17 00:00:00 2001
From: Anson Huang <b20788@freescale.com>
Date: Wed, 22 Jul 2015 00:40:00 +0800
Subject: [PATCH 0648/5242] MLK-11268-2 ARM: imx: disconnect vddhigh and
 vddsnvs in dsm for imx6ul

commit  17936e98a2ef4d5c1d448f3f7cc8cb5ff02cbb20 from
https://source.codeaurora.org/external/imx/linux-imx.git

Same as i.MX6SX, need to disconnect vddhigh and vddsnvs
in DSM on i.MX6UL, they have same design.

Signed-off-by: Anson Huang <b20788@freescale.com>
(cherry picked from commit f0c63b894a60512318481cb8a7b0777cdb7c46ab)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/anatop.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-imx/anatop.c b/arch/arm/mach-imx/anatop.c
index 058bf8b..c51fb15 100644
--- a/arch/arm/mach-imx/anatop.c
+++ b/arch/arm/mach-imx/anatop.c
@@ -46,7 +46,7 @@
 #define BM_ANADIG_ANA_MISC0_V3_STOP_MODE_CONFIG	0xc00
 /* Below MISC0_DISCON_HIGH_SNVS is only for i.MX6SL */
 #define BM_ANADIG_ANA_MISC0_DISCON_HIGH_SNVS	0x2000
-/* i.MX6SX DISCON_HIGH_SNVS is changed to bit 12 */
+/* Since i.MX6SX, DISCON_HIGH_SNVS is changed to bit 12 */
 #define BM_ANADIG_ANA_MISC0_V2_DISCON_HIGH_SNVS	0x1000
 #define BM_ANADIG_USB_CHRG_DETECT_CHK_CHRG_B	0x80000
 #define BM_ANADIG_USB_CHRG_DETECT_EN_B		0x100000
@@ -86,7 +86,7 @@ static inline void imx_anatop_enable_2p5_pulldown(bool enable)
 
 static inline void imx_anatop_disconnect_high_snvs(bool enable)
 {
-	if (cpu_is_imx6sx())
+	if (cpu_is_imx6sx() || cpu_is_imx6ul())
 		regmap_write(anatop, ANADIG_ANA_MISC0 +
 			(enable ? REG_SET : REG_CLR),
 			BM_ANADIG_ANA_MISC0_V2_DISCON_HIGH_SNVS);
@@ -116,7 +116,7 @@ void imx_anatop_pre_suspend(void)
 
 	imx_anatop_enable_fet_odrive(true);
 
-	if (cpu_is_imx6sl() || cpu_is_imx6sx())
+	if (cpu_is_imx6sl() || cpu_is_imx6sx() || cpu_is_imx6ul())
 		imx_anatop_disconnect_high_snvs(true);
 }
 
@@ -140,7 +140,7 @@ void imx_anatop_post_resume(void)
 
 	imx_anatop_enable_fet_odrive(false);
 
-	if (cpu_is_imx6sl() || cpu_is_imx6sx())
+	if (cpu_is_imx6sl() || cpu_is_imx6sx() || cpu_is_imx6ul())
 		imx_anatop_disconnect_high_snvs(false);
 
 }
-- 
1.7.9.5

